rueidiscompat/command.go

```diff
@@ -2109,6 +2112,134 @@ func (cmd *ClusterSlotsCmd) Result() ([]ClusterSlot, error) {
 	return cmd.val, cmd.err
 }
 
+func newClusterShardsCmd(res rueidis.RedisResult) *ClusterShardsCmd {
+	arr, err := res.ToArray()
+	if err != nil {
+		return &ClusterShardsCmd{err: err}
+	}
+	val := make([]ClusterShard, 0, len(arr))
+	for _, v := range arr {
+		dict, err := v.ToMap()
+		if err != nil {
+			return &ClusterShardsCmd{err: err}
+		}
+		var (
+			start, end int64
+			nodesArr   []ClusterNode
+		)
+		{
+			slots, ok := dict["slots"]
+			if !ok {
+				return &ClusterShardsCmd{err: errors.New("slots not found")}
+			}
+			arr, err := slots.ToArray()
```

Comment:
We can use `AsIntSlice` instead and ignore the `err`.
```suggestion
			arr, _ := slots.AsIntSlice()
```

Reply:
The `slots` in `RedisMessage`  are of type `blob string` therefore `AsIntSlice()` doesn't work.

Reply:
That's weird. Examples in the document are integers.

Reply:
It's based on the provided docker stack.
```
127.0.0.1:7010> CLUSTER SHARDS
1) 1) "slots"
   2) 1) "10923"
      2) "16383"
   3) "nodes"
   ....
```